### PR TITLE
DevWs for TechniSat Digital GmbH containing containing TechniSat Shutter

### PR DESCRIPTION
**Pull request by SmartThings**

### CODE REVIEW

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import groovy.json.JsonOutput
+
+metadata {
+	definition (name: "TechniSat Roller shutter switch", namespace: "TechniSat", author: "TechniSat", vid:"generic-shade",
+				mnmn: "SmartThings", runLocally: true, minHubCoreVersion: '000.017.0012',
+				executeCommandsLocally: false) {
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy:19]: remove `runLocally`, `minHubCoreVersion` and `executeCommandsLocally` flags

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+	def version = commandClassVersions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		log.debug "Parsed Crc16Encap into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract CRC16 command from $cmd"
+	}
+}
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy:276]: Is this required by the device? If not, please remove this method.

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using CRC16 Encapsulation, command: $cmd"
+	zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+}
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy:286]: Is this required by the device? If not, please remove this method.

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy (hunk 1 above)
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy:19]: This `vid` doesn't support Power Meter and Energy Meter.
You can create your own presentation using SmartThings CLI.
Please see https://smartthings.developer.samsung.com/docs/devices/device-profile.html#Device-Presentation and https://github.com/SmartThingsCommunity/smartthings-cli#smartthings-capabilitiespresentation-id-version

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+				mnmn: "SmartThings", runLocally: true, minHubCoreVersion: '000.017.0012',
+				executeCommandsLocally: false) {
+		capability "Window Shade"
+		capability "Window Shade Preset"
+		capability "Power Meter"
+		capability "Energy Meter"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Configuration"     
+        capability "Switch Level"   // until we get a Window Shade Level capability
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy:27]: use `"Window Shade Level"` capability instead

File: devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+
+		fingerprint mfr: "0299", prod: "0005", model: "1A93", deviceJoinName: "TechniSat Shutter"
```
Comment (PKacprowiczS) [inline devicetypes/technisat/technisat-roller-shutter-switch.src/technisat-roller-shutter-switch.groovy:29]: Change `deviceJoinName` to `TechniSat Window Treatment`

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]
@NArlt Please use consistent indentation (preferably tabs). Also remember to follow style guidelines from https://github.com/greens/DTH_development (like spaces after `if`).
Some parts of code give me impression that you haven't verified at all if device works correctly with this DTH. Please provide requested changes and actually test device with SmartThings ecosystem.

**Comment by NArlt** [comment]
Hi @PKacprowiczS ,
sorry for the late reply. 
According to the VID of the DTH: I know the currently available VIDs do not support energy and power metering. TechniSat tested the DTH and is fine without the metering in the view for now, but they will maybe decide to add it later. So I implemented the support already. If this is a Problem I will remove the capabilities. I will fix the other requested changes in a new PR and link it here.

BR NArlt

**Comment by NArlt** [comment]
Created new PR #67400

**Comment by k-morawski** [comment]
> #67400

Thank you @NArlt , we will review your new PR: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/67400. Could you please close https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/66240?

Many thanks!

**Comment by NArlt** [comment]
Hi @k-morawski ,
just to play it safe: This PR is created with DevWs. So I have no permission to simply close this PR. To close it, I have to cancel the publish request in DevWs. Canceling the request will not abort the publish process?

BR NArlt

**Comment by k-morawski** [comment]
> Hi @k-morawski ,
> just to play it safe: This PR is created with DevWs. So I have no permission to simply close this PR. To close it, I have to cancel the publish request in DevWs. Canceling the request will not abort the publish process?
> 
> BR NArlt

Thank you for explanation @NArlt . I wasn't aware about it. Please leave PR https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/66240 open. Thank you!

**Comment by k-morawski** [comment]
I am closing this pull request.

**Pull request closed without merging**